### DISCUSSION TITLE
Upgrading to 1.3 and beyond?

### DISCUSSION THREAD

**Discussion by TimHeckel** [mysticaltech/terraform-hcloud-kube-hetzner#214] [Q&A] [answered]
Hi everyone - I was simply wondering what the recommended approach is to upgrading the cluster. I have it running with the pre-release version from early February, and I know lots of improvements have been put in place. Is it wise to simply start over, or is there an upgrade strategy that you'd suggest?

Also going forward, is there a way to upgrade to the next version with some command? (Say I wanted to move from 1.2 to 1.3).

Thanks again for this amazing work!

**Comment by mysticaltech** [comment, marked as answer]
Hey @TimHeckel, genuine concern man. For the moment, we do not have an easy way to upgrade. But you could use [Velero](https://github.com/vmware-tanzu/velero) a nice tool by VMWare to do that! 

I haven't tried it myself, but it seems logical that it would work.

**Reply by mysticaltech** [reply]
Here's a demo, seems pretty straightforward. Basically, make a backup with Velero of your old cluster and then restore it to your new cluster. https://www.youtube.com/watch?v=IZlwKMoqBqE

**Reply by otavio** [reply]
Now 1.3 is out, does upgrade capabilities being considered?

**Reply by mysticaltech** [reply]
Hello @otavio, the easiest thing if you have an old version, is to just use Velero https://github.com/vmware-tanzu/velero to do the migration from the old setup to the new. Now that said, both MicroOS and k3s stay the exact same, it's just the terraform deploy code that has changed in 1.3.

**Comment by TimHeckel** [comment]
Hi @mysticaltech! Thanks for the tip - I will give this a try when I get a chance :)
